misc/cf-support

```diff
@@ -0,0 +1,189 @@
+.SH PROMISE THEORY
+CFEngine is built on principles from promise theory, proposed by Mark Burgess in 2004. Promise theory is a model of voluntary cooperation between individual, autonomous actors or agents who publish their intentions to one another in the form of promises. A promise is a declaration of intent whose purpose is to increase the recipient's certainty about a claim of past, present or future behaviour. For a promise to increase certainty, the recipient needs to trust the promiser, but trust can also be built on the verification that previous promises have been kept, thus trust plays a symbiotic relationship with promises.
+.br
+For an introduction to promise theory, please see http://arxiv.org/abs/0810.3294/
+.SH AVAILABILITY
+cf-support is part of CFEngine.
+.br
+Binary packages may be downloaded from http://cfengine.com/downloads/.
+.br
+The source code is available at http://github.com/cfengine/
+.SH BUGS
+Please see the public bug-tracker at http://bug.cfengine.com/.
+.br
+GitHub pull-requests may be submitted to http://github.com/cfengine/core/.
+.SH "SEE ALSO"
+.BR cf-promises (8),
+.BR cf-agent (8),
+.BR cf-serverd (8),
+.BR cf-execd (8),
+.BR cf-monitord (8),
+.BR cf-runagent (8),
+.BR cf-key (8)
+.SH AUTHOR
+Northern.tech AS
+EOF
+  exit 0
+fi
+
+[ "$1" = "--yes" ] || [ "$1" = "-y" ] && non_interactive=1
+
+if [ "$EUID" -ne 0 ]; then
+  echo "$0 must be run as root"
+  exit 1
+fi
+
+if [ "$(command -v gzip >/dev/null)" ]; then
+  echo "Please install gzip. This is required in order to minimize size of support collection."
+  exit 1
+fi
+
+if [ $non_interactive -eq 0 ]; then
+  read -rp "If you have it, please enter your support case number: " case_number
+fi
+case_number="${case_number:-NA}"
+timestamp=$(date +%F-%H%M)
+collection="cfengine_support_case_$case_number-$(hostname -f)-$timestamp"
+
+tmpdir="$(mktemp -d)/$collection"
+export tmpdir
+mkdir -p "$tmpdir"
+
+export info_file="$tmpdir/system-info.txt"
+
+function file_add
+{
+  local filename
+  filename="$(basename "$1")"
+  if [ -f "$1" ]; then
+    cp "$1" "$tmpdir"/"$filename"
+    echo "Added $1"
+  else
+    echo "$1 not found" >> "$info_file"
+  fi
+}
+export -f file_add
+
+function gzip_add
+{
+  local filename
+  filename="$(basename "$1")"
+  if [ -f "$1" ]; then
+    gzip -c "$1" > "$tmpdir"/"$filename".gz
+    echo "Added compressed $1"
+  else
+    echo "$1 not found" >> "$info_file"
+  fi
+}
+export -f gzip_add
+
+function log_cmd
+{
+  echo "=== $1 ===" >> "$info_file"
+  eval "$1" >> "$info_file" 2>&1 || true
+  echo >> "$info_file" # one newline for easier to read output
+  echo "Captured output of command $1"
+}
+export -f log_cmd
+
+log_cmd "hostname -f"
+log_cmd "uname -a"
+[ -f "/etc/os-release" ] && log_cmd "cat /etc/os-release"
+log_cmd "$BINDIR/cf-promises -V"
+log_cmd "free -m"
+log_cmd "df -h"
+log_cmd "ps auwwx"
+log_cmd "top -b -H -c -n1"
+log_cmd "netstat -ie"
+netstat -ie >/dev/null 2>/dev/null || log_cmd "ifconfig"
+log_cmd "ifconfig"
```

Comment:
Double `ifconfig` here.

Reply:
oh. right. I could be less "tricky" and just run both. :)